### PR TITLE
feat: Add segment tracking for bulk enrollment

### PR DESCRIPTION
**Pull request by taliaedX**
***Background***
This adds bulk enrollment specific segment tracking to the enrollment action that occurs only via bulk enrollment.

***Screenshot***
Screen Shot 2021-07-01 at 3 33 31 PM

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/api/v1/views.py [hunk 1]
```diff
@@ -313,6 +314,7 @@ def enroll_learners_in_courses(self, request, pk):
                 result.pop('user') for result in results['successes'] if result['course_run_key'] == course_run
             }
             LOGGER.info("Successfully bulk enrolled learners: {}".format(pending_users | existing_users))
+            track_enrollment('customer-admin-enrollment', request.user.id, course_run)
```
Comment (binodpant) [inline enterprise/api/v1/views.py:317]: if possible we should add a test for this tracking call too

**Review by binodpant — APPROVED** [review]
well yeah this one looks nice and easy! yeah let's find out more about the source field requirements for pending and regular enrollments. but this meets the criteria for AC1 in the ticket so I think we can go in. Just made a suggestion about unit test.

File: enterprise/api/v1/views.py [hunk 2]
```diff
@@ -312,14 +313,16 @@ def enroll_learners_in_courses(self, request, pk):
...
             }
-            LOGGER.info("Successfully bulk enrolled learners: {}".format(pending_users | existing_users))
-            if serializer.validated_data.get('notify'):
-                enterprise_customer.notify_enrolled_learners(
-                    catalog_api_user=request.user,
-                    course_id=course_run,
-                    users=pending_users | existing_users,
-                    admin_enrollment=True,
-                )
+            if len(pending_users | existing_users) > 0:
```
Comment (alex-sheehan-edx) [inline enterprise/api/v1/views.py:316]: Assuming this is to prevent the empty log lines, nice!

**Review by alex-sheehan-edx — APPROVED** [review]
nice tests, lgtm

File: enterprise/api/v1/views.py (hunk 2 above)
Comment (taliaedX) [reply enterprise/api/v1/views.py:316]: yes that is this bit!

**Pull request merged**